theano/gpuarray/dnn_redux.c

```diff
@@ -61,11 +61,6 @@ int APPLY_SPECIFIC(dnn_redux)(PyGpuArrayObject *input,
   static float fbeta = 0.0f;
   static double dbeta = 0.0;  
 
-  if (!GpuArray_IS_C_CONTIGUOUS(&input->ga)) {
```

Comment:
Indeed, I checked the doc as well, and nothing indicates that negative strides, or even overlapping strides (!) would be an issue, as those restrictions are indicated for other algos (LCN for instance).
I'd still feel safer if we have a test that the library indeed behaves according to the doc...